pkg/api/volume_req.go

```diff
@@ -27,16 +38,31 @@ type SubvolReq struct {
 "create-brick-dir" : if brick dir is not present, create it
 */
 type VolCreateReq struct {
-	Name         string            `json:"name,omitempty"`
-	Transport    string            `json:"transport,omitempty"`
-	Subvols      []SubvolReq       `json:"subvols"`
-	Options      map[string]string `json:"options,omitempty"`
-	Force        bool              `json:"force,omitempty"`
-	Advanced     bool              `json:"advanced,omitempty"`
-	Experimental bool              `json:"experimental,omitempty"`
-	Deprecated   bool              `json:"deprecated,omitempty"`
-	Metadata     map[string]string `json:"metadata,omitempty"`
-	Flags        map[string]bool   `json:"flags,omitempty"`
+	Name                    string            `json:"name,omitempty"`
+	Transport               string            `json:"transport,omitempty"`
+	Subvols                 []SubvolReq       `json:"subvols"`
+	Options                 map[string]string `json:"options,omitempty"`
+	Force                   bool              `json:"force,omitempty"`
+	Advanced                bool              `json:"advanced,omitempty"`
+	Experimental            bool              `json:"experimental,omitempty"`
+	Deprecated              bool              `json:"deprecated,omitempty"`
+	Metadata                map[string]string `json:"metadata,omitempty"`
+	Flags                   map[string]bool   `json:"flags,omitempty"`
+	Size                    uint64            `json:"size"`
+	DistributeCount         int               `json:"distribute,omitempty"`
+	ReplicaCount            int               `json:"replica,omitempty"`
+	ArbiterCount            int               `json:"arbiter,omitempty"`
+	DisperseCount           int               `json:"disperse,omitempty"`
+	DisperseRedundancyCount int               `json:"disperse-redundancy,omitempty"`
+	DisperseDataCount       int               `json:"disperse-data,omitempty"`
+	SnapshotEnabled         bool              `json:"snapshot,omitempty"`
+	SnapshotReserveFactor   float64           `json:"snapshot-reserve-factor,omitempty"`
+	LimitPeers              []string          `json:"limit-peers,omitempty"`
+	LimitZones              []string          `json:"limit-zones,omitempty"`
+	ExcludePeers            []string          `json:"exclude-peers,omitempty"`
+	ExcludeZones            []string          `json:"exclude-zones,omitempty"`
+	SubvolZonesOverlap      bool              `json:"subvolume-zones-overlap,omitempty"`
+	SubvolType              string            `json:"subvolume-type,omitempty"`
```

Comment:
These two structs are now huge. I think it would be a good idea to group all the IVP related members in to their own structs and add those structs as members of the main structs.

Reply:
I feel input API will not be intuitive if we make it as sub level.

Reply:
I think otherwise. But we can finalize the request structure later.